helix-rest/src/main/java/org/apache/helix/rest/server/resources/AbstractResource.java

```diff
@@ -98,6 +98,11 @@ protected Response serverError() {
     return Response.serverError().build();
   }
 
+  protected Response serverError(String errorMsg) {
+    return Response.status(Response.Status.INTERNAL_SERVER_ERROR).entity(errorMsgToJson(errorMsg))
```

Comment:
Let's put it this way Response.serverError().entity(errorMsgToJson(errorMsg)).build() to be consistent with the others serverError methods.

Reply:
serverError() contains build(). We do not want to build before we set the entity. It's simply unnecessary. The consistency you're looking for is with the notFound() function with the same parameter.

Reply:
serverError() doesn't contain build(), it simply returns Response.status(Response.Status.INTERNAL_SERVER_ERROR). There's no fundamental difference, it's fine if you keep it this way.